sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -83,122 +93,172 @@ private boolean looksLikeSqlObject(Class<?> extensionType) {
      */
     @Override
     public <E> E attach(Class<E> extensionType, HandleSupplier handle) {
-        Map<Method, Handler> handlers = methodHandlersFor(
-                extensionType,
-                handle.getConfig(Handlers.class),
-                handle.getConfig(HandlerDecorators.class));
-
         ConfigRegistry instanceConfig = handle.getConfig().createCopy();
 
-        for (Class<?> iface : extensionType.getInterfaces()) {
-            forEachConfigurer(iface, (configurer, annotation) ->
-                configurer.configureForType(instanceConfig, annotation, extensionType));
-        }
-        forEachConfigurer(extensionType, (configurer, annotation) ->
-                configurer.configureForType(instanceConfig, annotation, extensionType));
-
-        InvocationHandler invocationHandler = createInvocationHandler(extensionType, instanceConfig, handlers, handle);
-        return extensionType.cast(
-                Proxy.newProxyInstance(
-                        extensionType.getClassLoader(),
-                        new Class[] {extensionType},
-                        invocationHandler));
-    }
+        SqlObjectInitData data = sqlObjectCache.get(extensionType, handle.getConfig());
+        data.instanceConfigurer.apply(instanceConfig);
 
-    private Map<Method, Handler> methodHandlersFor(Class<?> sqlObjectType, Handlers registry, HandlerDecorators decorators) {
-        return handlersCache.computeIfAbsent(sqlObjectType, type -> {
-            final Map<Method, Handler> handlers = new HashMap<>();
-
-            handlers.putAll(handlerEntry((t, a, h) ->
-                    sqlObjectType.getName() + '@' + Integer.toHexString(t.hashCode()),
-                Object.class, "toString"));
-            handlers.putAll(handlerEntry((t, a, h) -> t == a[0], Object.class, "equals", Object.class));
-            handlers.putAll(handlerEntry((t, a, h) -> System.identityHashCode(t), Object.class, "hashCode"));
-            handlers.putAll(handlerEntry((t, a, h) -> h.getHandle(), SqlObject.class, "getHandle"));
+        if (data.concrete) {
             try {
-                handlers.putAll(handlerEntry((t, a, h) -> null, sqlObjectType, "finalize"));
-            } catch (IllegalStateException expected) {
-                // optional implementation
+                SqlObjectInitData.INIT_DATA.set(data);
+                return extensionType.cast(
+                    Class.forName(extensionType.getPackage().getName() + "." + extensionType.getSimpleName() + "Impl")
```

Comment:
let's make `"Impl"` a constant on a new class that handles these naming concerns?